p2p/net/nat/nat.go

```diff
@@ -34,7 +34,7 @@ type entry struct {
 var discoverGateway = nat.DiscoverGateway
 
 // DiscoverNAT looks for a NAT device in the network and returns an object that can manage port mappings.
-func DiscoverNAT(ctx context.Context) (*NAT, error) {
+func DiscoverNAT(ctx context.Context, userAgent string) (*NAT, error) {
```

Comment:
Do we want to change the interface of this public method?

Reply:
What are the alternatives?

Reply:
Introducing a new func.

Reply:
If we create a new function, we need to designate a default userAgent somewhere in this file. The default value is set in another file.
So this constructor has no sense without the userAgent string.

Reply:
The default should be in this file. That was the old behavior. If users want to change it they can call the new function (maybe call it `DiscoverNATWithUserAgent`).

This change as is is a breaking API change. I like to avoid them unless necessary. I don't think it's necessary here.

Reply:
@MarcoPolo Okay. Then what to do with this one?
https://github.com/libp2p/go-libp2p/pull/2310/files#diff-a5aa28aae96d02e4316747f7950909f36d447f7833e5a70121c3808852e5e440R80

What to pass in this line? What will be the default value here?

Reply:
And then what will be in this place? "if" and two constructors? 
https://github.com/libp2p/go-libp2p/pull/2310/files#diff-a5aa28aae96d02e4316747f7950909f36d447f7833e5a70121c3808852e5e440R117

Reply:
@MarcoPolo Okay. Why shouldn't there be an incoming userAgent parameter in this function? After all, we cannot communicate with NAT without userAgent.

Reply:
>Why shouldn't there be an incoming userAgent parameter in this function?

Because it's a breaking API change. Users of this function, when they update go-libp2p will need to update their usage of this function and pass in a userAgent. 

It'd be nicer if we add another function `DiscoverNATWithUserAgent` and `NatManager` switches based on whether it has a user configured UserAgent or not

Reply:
@sukunrt Okay. I fixed it.